module-bluetooth/Bluetooth/Device.cpp

```diff
@@ -0,0 +1,78 @@
+#include "Device.hpp"
+#include <btstack_util.h>
+
+Devicei::Devicei(std::string name) : Device(std::move(name))
+{
+    pageScanRepetitionMode={};
+    clockOffset={};
+    classOfDevice={};
+    state=DEVICE_STATE::REMOTE_NAME_FAILURE;
+    deviceState=DeviceState::Unknown;
+    isPairingSSP = false;
+
+    memset(address, 0, sizeof(address));
+}
+
+Devicei::Devicei(bd_addr_t &address) : Devicei()
+{
+    setAddress(&address);
+}
+
+Devicei &Devicei::operator=(const Devicei &d)
+{
+    if (&d == this) {
+        return *this;
+    }
+    setAddress(&d.address);
+    setName(d.name.data());
+    pageScanRepetitionMode = d.pageScanRepetitionMode;
+    clockOffset            = d.clockOffset;
+    classOfDevice          = d.classOfDevice;
+    state                  = d.state;
+    deviceState            = d.deviceState;
+    isPairingSSP           = d.isPairingSSP;
+    return *this;
+}
+
+Devicei::Devicei(const Devicei &d) : Devicei(d.name.data())
+{
+    operator=(d);
+}
+
+Devicei &Devicei::operator=(Devicei &&d) noexcept
+{
+    setAddress(&d.address);
+    setName(d.name.data());
+    pageScanRepetitionMode = d.pageScanRepetitionMode;
+    clockOffset            = d.clockOffset;
+    classOfDevice          = d.classOfDevice;
+    state                  = d.state;
+    deviceState            = d.deviceState;
+    isPairingSSP           = d.isPairingSSP;
+    return *this;
+}
+
+Devicei::Devicei(Devicei &&d) noexcept : Devicei(d.name.data())
+{
+    operator=(d);
+}
+
+void Devicei::setAddress(bd_addr_t *addr)
+{
+    memcpy(&address, addr, sizeof address);
```

Comment:
I think that there's a function in BT stack to copy the address, but I won't be surprised if it was just a wrapped `memcpy`

Reply:
I just moved the header code to cpp in this case :)